src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml

```diff
@@ -51,6 +51,11 @@
             <tag name="controller.argument_value_resolver" priority="-150" />
         </service>
 
+        <service id="argument_resolver.not_tagged_controller" class="Symfony\Component\HttpKernel\Controller\ArgumentResolver\ControllerNotRegisteredAsServiceValueResolver">
```

Comment:
I would only enable this resolver when `debug` is `true`.

Reply:
I added the registration of the `NotTaggedControllerValueResolver` resolver service to the `Symfony\Component\HttpKernel\DependencyInjection\ControllerArgumentValueResolverPass`.